proto/osmosis/lockup/lock.proto

```diff
@@ -8,74 +8,82 @@ import "cosmos/base/v1beta1/coin.proto";
-// at a specific time. It stores owner, duration, unlock time and the amount of
-// coins locked.
+// PeriodLock is a single unit of lock by period defined by the x/lockup module.
+// It's a record of locked coin at a specific time. It stores owner, duration,
+// unlock time and the amount of coins locked. A state of a period lock is
+// created upon lock creation, and deleted once the lock has been matured after
+// the `duration` has passed since unbonding started.
 message PeriodLock {
+  // ID is the unique id of the lock.
+  // The ID of the lock is decided upon lock creation, incrementing by 1 for
+  // every lock.
   uint64 ID = 1;
+  // Owner is the account address of the lock owner.
+  // Only the owner can modify the state of the lock.
   string owner = 2 [ (gogoproto.moretags) = "yaml:\"owner\"" ];
+  // Duration is the time needed for a lock to be matured after unlocking has
+  // started.
   google.protobuf.Duration duration = 3 [
     (gogoproto.nullable) = false,
     (gogoproto.stdduration) = true,
     (gogoproto.jsontag) = "duration,omitempty",
     (gogoproto.moretags) = "yaml:\"duration\""
   ];
+  // End Time refers to the time lock would be matured and deleted.
+  // This value is first initialized when an unlock has started for the lock,
+  // end time being block time + duration.
   google.protobuf.Timestamp end_time = 4 [
     (gogoproto.stdtime) = true,
     (gogoproto.nullable) = false,
     (gogoproto.moretags) = "yaml:\"end_time\""
   ];
+  // Coins are the coins locked within the lock, kept in the module account.
   repeated cosmos.base.v1beta1.Coin coins = 5 [
     (gogoproto.nullable) = false,
     (gogoproto.castrepeated) = "github.com/cosmos/cosmos-sdk/types.Coins"
   ];
 }
 
+// LockQueryType defines the type of the lock query, query being
+// either by duration or start time of the lock.
 enum LockQueryType {
   option (gogoproto.goproto_enum_prefix) = false;
 
-  // Queries for locks that are longer than a certain duration
   ByDuration = 0;
-  // Queries for lockups that started before a specific time
   ByTime = 1;
 }
 
+// QueryCondition is a struct used for querying locks upon different conditions.
+// Duration field and timestamp fields could be optional, depending on the
+// LockQueryType.
 message QueryCondition {
   // type of lock query, ByLockDuration | ByLockTime
   LockQueryType lock_query_type = 1;
   // What token denomination are we looking for lockups of
   string denom = 2;
-  // valid when query condition is ByDuration
+  // Locks with longer duration than the specified duration would be queried.
+  // Duration field should not be nil when lock query type is by duration.
   google.protobuf.Duration duration = 3 [
     (gogoproto.stdduration) = true,
     (gogoproto.nullable) = false,
     (gogoproto.moretags) = "yaml:\"duration\""
   ];
-  // valid when query condition is ByTime
+  // Locks started before specified timestamp would be queried.
```

Comment:
I don't understand why this timestamp is relevant specifically to queries? Could we paraphrase or elaborate on the description, please?
```suggestion
  // Timestamp Locks started before specified timestamp would be queried.
```

Reply:
updated comments on this, lmk what you think!